File: programmers/체육복.py
def solution(n, lost, reserve):
    answer = 0
    student = [1] * (n+2)
    
    for l in lost:
        student[l] -= 1
    for r in reserve:
        student[r] += 1

    for i in range(1,n+1):
        if student[i] > 1:
            if student[i-1] == 0:
                student[i] -= 1
                student[i-1] += 1
            elif student[i+1] == 0:
                student[i] -= 1
                student[i+1] += 1
    
    for i in range(1,n+1):
        if student[i] > 0:
            answer += 1
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
class Solution {
    
    static final int NORMAL  = 0;
    static final int LOST = 1;
    static final int RESERVE = 2;
    
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        
        int[] student = new int[n+1];
        for (int i = 0; i < lost.length; i++) {
            student[lost[i]] = LOST;
            answer--;
        }
        
        for (int i = 0; i < reserve.length; i++) {
            if (student[reserve[i]] == LOST) {
                student[reserve[i]] = NORMAL;
                answer++;
            } else {
                student[reserve[i]] = RESERVE;
            }
        }
        
        for (int i = 1; i < n+1; i++) {
            if (student[i] == LOST && student[i-1] == RESERVE)  {
                student[i-1] = NORMAL;
                student[i] = NORMAL;
                answer++;
            } else if (i < n && student[i] == LOST && student[i+1] == RESERVE) {
                student[i+1] = NORMAL;
                student[i] = NORMAL;
                answer++;
            }
        }
        
        return answer;
    }
}